lib/beaker/test_suite_result.rb

```diff
@@ -0,0 +1,259 @@
+      else
+        test_case.test_status
+      end
+      "  Test Case #{test_case.path} #{test_reported}"
+    end
+
+    # Writes Junit XML of this {TestSuiteResult}
+    #
+    # @param [String] xml_file      Path to the XML file (from Beaker's running directory)
+    # @param [String] file_to_link  Path to the paired file that should be linked
+    #                               from this one (this is relative to the XML
+    #                               file itself, so it would just be the different
+    #                               file name if they're in the same directory)
+    # @param [Boolean] time_sort    Whether the test results should be output in
+    #                               order of time spent in the test, or in the
+    #                               order of test execution (default)
+    #
+    # @return nil
+    # @api private
+    def write_junit_xml(xml_file, file_to_link = nil, time_sort = false)
+      stylesheet = File.join(@options[:project_root], @options[:xml_stylesheet])
+
+      begin
+        LoggerJunit.write_xml(xml_file, stylesheet) do |doc, suites|
+
+          meta_info = suites.add_element(REXML::Element.new('meta_test_info'))
+          unless file_to_link.nil?
+            time_sort ? meta_info.add_attribute('page_active', 'performance') : meta_info.add_attribute('page_active', 'execution')
+            meta_info.add_attribute('link_url', file_to_link)
+          else
+            meta_info.add_attribute('page_active', 'no-links')
+            meta_info.add_attribute('link_url', '')
+          end
+
+          suite = suites.add_element(REXML::Element.new('testsuite'))
+          suite.add_attributes(
+            [
+              ['name' , @name],
+              ['tests', test_count],
+              ['errors', errored_tests],
+              ['failures', failed_tests],
+              ['skipped', skipped_tests],
+              ['pending', pending_tests],
+              ['total', @total_tests],
+              ['time', "%f" % (stop_time - start_time)]
+          ])
+          properties = suite.add_element(REXML::Element.new('properties'))
+          @options.each_pair do |name,value|
+            property = properties.add_element(REXML::Element.new('property'))
+            property.add_attributes([['name', name], ['value', value.to_s || '']])
+          end
+
+          test_cases_to_report = @test_cases
+          test_cases_to_report = @test_cases.sort { |x,y| y.runtime <=> x.runtime } if time_sort
+          test_cases_to_report.each do |test|
+            item = suite.add_element(REXML::Element.new('testcase'))
+            item.add_attributes(
+              [
+                ['classname', File.dirname(test.path)],
+                ['name', File.basename(test.path)],
+                ['time', "%f" % test.runtime]
+              ])
+
+            test.exports.each do |export|
+              export.keys.each do |key|
+                item.add_attribute(key.to_s.tr(" ", "_"), export[key])
```

Comment:
i better be thumbs up with this.  you are fixing my change 👼 
https://github.com/puppetlabs/beaker/blame/master/lib/beaker/test_suite.rb#L213